js/everything.js

```diff
@@ -428,24 +489,104 @@ function anchorURLs(text) {
     return text.replace(re, "<a href=\"$1\" target=\"_blank\">$1</a>");
 }
 
-function wrapParseYouTubeVideoID(url) {
+function wrapParseVideoID(url, callback) {
     if (currentVideoID && url === currentVideoID) {
         // We have already determined the video id
-        return currentVideoID;
-    }
-
-    var info = parseYoutubeVideoID(url);
-
-    if (info.id) {
-        currentVideoID = info.id;
-        ga("send", "event", "video ID format", info.format);
-        return info.id;
+        callback.success(currentVideoID);
     }
     else {
-        errorMessage.show("Failed to parse the video ID.");
+        // First try parse Soundcloud id from URL.
+        parseSoundcloudVideoID(
+            url,
+            soundcloudClientID,
+            {
+                success: function(info) {
+                    // Success parsed ID from Soundcloud URL.
+                    ga("send", "event", "video ID format", info.format);
+                    currentVideoID = info.id;
+                    callback.success(info.id);
+                },
+                fail: function(info) {
+                    // Failed, so next try parse Youtube ID from URL.
+                    info = parseYoutubeVideoID(url);
+                    if (info.id) {
+                        // Success parsed Youtube ID.
+                        currentVideoID = info.id;
+                        callback.success(info.id);
+                    }
+                    else {
+                        // No Soundcloud or Youtube ID.
+                        callback.fail();
+         //               errorMessage.show("Failed to parse the video ID.");
+                    }
+                }
+            });
     }
 }
 
+function verifyYoutubeID(id, callback) {
+    $.ajax({
+        url: "https://www.googleapis.com/youtube/v3/videos",
+        dataType: "json",
+        data: {
+            key: youTubeDataApiKey,
+            part: "snippet",
+            fields: "items/snippet/description",
+            id: id
+        },
+        success: function(data) {
+            if (data.items.length === 0) {
+                callback.fail(false);
```

Comment:
Why are we passing in false/true in all of these `fail()`/`success()` functions? Aren't we the creator & consumer of them? I don't see where the param is used anywhere - am I missing something?